geopaparazzi.app/src/eu/hydrologis/geopaparazzi/maps/overlays/GeopaparazziOverlay.java

```diff
@@ -506,11 +514,62 @@ protected void drawOverlayBitmap(Canvas canvas, Point drawPosition, Projection p
             visibleItems.addAll(visibleItemsRedraw);
             visibleItemsRedraw.addAll(visibleItemsTemp);
         }
-
-        /*
-         * gps logging track
-         */
-        if (gpsLoggingStatus == GpsLoggingStatus.GPS_DATABASELOGGING_ON) {
+    }
+    /**
+     * Draws Ways (gps+gpx Log-Tracks)
+     * (called from drawOverlayBitmap )
+     *
+     * @param canvas
+     * @param drawPosition
+     * @param projection
+     * @param drawZoomLevel
+     * @return nothing
+     */
+    private void drawWays(Canvas canvas, Point drawPosition, Projection projection, byte drawZoomLevel) {
+        int numberOfWays = waySize();
+        for (int wayIndex = 0; wayIndex < numberOfWays; ++wayIndex) {
+            if (stopDrawing()) {
+                // stop working
+                return;
+            }
+            // get the current way
+            OverlayWay overlayWay = createWay(wayIndex);
+            if (overlayWay == null) {
+                continue;
+            }
+            // make sure that the current way has way nodes
+            if (overlayWay.wayNodes == null || overlayWay.wayNodes.length == 0) {
+                continue;
+            }
+            // make sure that the cached way node positions are valid
+            if (drawZoomLevel != overlayWay.cachedZoomLevel) {
+                for (int i = 0; i < overlayWay.cachedWayPositions.length; ++i) {
+                    for (int j = 0; j < overlayWay.cachedWayPositions[i].length; ++j) {
+                        overlayWay.cachedWayPositions[i][j] = projection.toPoint(overlayWay.wayNodes[i][j],
+                                overlayWay.cachedWayPositions[i][j], drawZoomLevel);
+                    }
+                }
+                overlayWay.cachedZoomLevel = drawZoomLevel;
+            }
+            drawWayPathOnCanvas(canvas, drawPosition, overlayWay);
+        }
+    }
+    /**
+     * Draws GpsLog  (while Logging is on)
+     * (called from drawOverlayBitmap )
+     *
+     * @param canvas
+     * @param drawPosition
+     * @param projection
+     * @param drawZoomLevel
+     * @return nothing
+     */
+    private void drawGpsLog(Canvas canvas, Point drawPosition, Projection projection, byte drawZoomLevel) {
+     if (stopDrawing()) {
+        // stop working
+        return;
```

Comment:
This does no longer work as it should. This return was meant for the method it was in before.